smartcontractkit/external-adapters-js
# Allow setting custom Ogmios URLs for ADA balance adapter

## Closes 26372

## Description

There is a node provider that wishes to provide a Cardano node using a URL with a path other than the root.  The Cardano Ogmios library does not support this out of the box.  This PR modifies their library to allow for this use case.

## Changes

- Only allow 2 required environment variables for adapter.  These are `WS_OGMIOS_URL` and `HTTP_OGMIOS_URL`.
- Port over Cardano Ogmios `createInteractionContext` function from the Cardano Ogmios client library and repurpose to allow setting custom URLs.



## Steps to Test

1. Remove all environment variables and set `WS_OGMIOS_URL` and `HTTP_OGMIOS_URL` environment variables
2. Fire request to adapter and verify a that you get a successful response.

## Quality Assurance
